stock_picking_auto_create_lot_qty/models/stock_picking.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (C) 2023 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _set_auto_lot(self):
+        """
+        Allows to be called either by button or through code
+        """
+        pickings = self.filtered(lambda p: p.picking_type_id.auto_create_lot)
+        lines = pickings.mapped("move_line_ids").filtered(
+            lambda x: (
+                not x.lot_id
+                and not x.lot_name
+                and x.product_id.tracking != "none"
+                and x.product_id.auto_create_lot
+            )
+        )
+        new_lines = pickings.mapped("move_line_ids").filtered(lambda x: (x.qty_done))
+        if not new_lines:
+            count_by_product = {}
+            for line in lines:
+                if line.product_id not in count_by_product:
+                    count_by_product[line.product_id] = line.product_uom_qty
+                else:
+                    count_by_product[line.product_id] += line.product_uom_qty
```

Comment:
An easier way of doing this would be

```suggestion
            count_by_product = dict.fromkeys(lines.mapped("product_id"), 0)
            for line in lines:
                count_by_product[line.product_id] += line.product_uom_qty
```

Reply:
your suggestion will work only in case were we already have any value in dict by this key. If we create key we need to set value first and then we can add something to this value.

Reply:
Isn't that what `dict.fromkeys` would do? Anyway, I thought it would be a bit cleaner, but it's not a big deal regardless.

Reply:
done